src/app/components/add-content/add-content.component.ts

```diff
@@ -0,0 +1,77 @@
+import {Component, OnInit, ViewChild, ElementRef, EventEmitter, Output} from '@angular/core';
+import {FormBuilder, FormGroup, Validators} from '@angular/forms';
+import {CategoryService} from '../../services/communibee-backend/category/category.service';
+import {CategoryModel} from '../../services/communibee-backend/category/category';
+import {ContentModel} from '../../services/communibee-backend/content/content';
+import {ContentService} from '../../services/communibee-backend/content/content.service';
+import {ArrBuff} from '../../services/utils/arr-buff.service';
+
+@Component({
+  selector: 'app-add-content',
+  templateUrl: './add-content.component.html',
+  styleUrls: ['./add-content.component.scss'],
+})
+export class AddContentComponent implements OnInit {
+  public addContentForm: FormGroup;
+  categories: CategoryModel[] = [];
+  content: ContentModel = {} as any;
+  isFileSelected = false;
+  fileErrorSize = false;
+  contentResult: ContentModel = {} as any;
+  @ViewChild('closeBtn') closeBtn: ElementRef;
+  @Output() contentTitleLoaded = new EventEmitter<string>();
+
+  constructor(private fb: FormBuilder,
+              private categoriesSrv: CategoryService,
+              private contentSrv: ContentService) {
+    this.initForm();
+  }
+
+  ngOnInit() {
+    this.categoriesSrv.getAll().then(categories_res => {
+      this.categories = categories_res;
+    });
+  }
+
+  initForm() {
+    this.addContentForm = this.fb.group({
+      title: ['', Validators.required],
+      information: ['', [Validators.required,
+        Validators.maxLength(200)]],
+      category: ['', Validators.required],
+    });
+  }
+
+  onFileChange(event) {
+    const fileSize = event.srcElement.files[0].size;
+    const maxFileSize = 1.048576e7; // 5MB
+
+    this.fileErrorSize = false;
+    if (fileSize > maxFileSize) {
+        this.fileErrorSize = true;
+    } else {
+      const reader = new FileReader();
+      if (event.target.files && event.target.files.length) {
+        const [file] = event.target.files;
+        reader.readAsArrayBuffer(file);

```

Comment:
Shouldn't it be called after `reader.onload = ....`?

Reply:
no. Onload is triggered after the FileReader (reader) finished reading the file.